src/main/java/gr8pefish/ironbackpacks/container/ContainerBackpack.java

```diff
@@ -31,16 +33,46 @@
     private final BackpackInfo backpackInfo;
     @Nonnull
     private final BackpackSize backpackSize;
-    @Nonnull
+    @Nullable
     private final ItemStack backpackStack;
+    @Nullable
+    private final TileEntityBackpack backpackTileEntity;
+
+    private final boolean usesItemStack;
+    private final boolean usesTileEntity;
+
     /**
      * Offhand used or not.
      */
     private int blocked = -1;
 
     // Constructor
 
-    public ContainerBackpack(@Nonnull ItemStack backpackStack, @Nonnull InventoryPlayer inventoryPlayer, @Nonnull EnumHand hand) {
+    /** Constructor for the TE */
+    public ContainerBackpack(@Nonnull InventoryPlayer inventoryPlayer, @Nonnull EnumHand hand, @Nonnull TileEntityBackpack backpackTileEntity) {
```

Comment:
I don't really like how this is handled, but I can't really give a better impl right now.

Reply:
Yeah, I was thinking of passing in some (IBackpack?) that delegated the methods inside to either the ItemStack or the TE, as I agree, it isn't super nice right now. But I ran into some issues with that and an overloaded constructor (with some non-optimal fields) seemed simpler for the time being.